artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -87,78 +87,84 @@ impl TimeArgs {
+            0 | 9 | 11.. => {
+                let mut message = br#"wrong number of arguments (given "#.to_vec();
+                message.extend(args.len().to_string().bytes());
+                message.extend_from_slice(b", expected 1..8)");
+                Err(ArgumentError::from(message).into())
+            }
+            1..=8 => {
+                // For 0..=7 params, we need to validate to_int
+                let mut result = Args::default();
+                for (i, arg) in args.iter().enumerate() {
+                    // The eighth parameter is never used, and thus no
+                    // conversion is needed
+                    if i == 7 {
+                        continue;
+                    }
 
-                let arg = to_int(interp, *arg)?;
-                // unwrap is safe since to_int gaurnatees a non nil Ruby::Integer
-                let arg: i64 = arg.try_convert_into::<Option<i64>>(interp)?.unwrap();
-
-                match i {
-                    0 => result.year = arg,
-                    1 => result.month = arg,
-                    2 => result.day = arg,
-                    3 => result.hour = arg,
-                    4 => result.minute = arg,
-                    5 => result.second = arg,
-                    6 => result.micros = arg,
-                    7 => {
-                        // NOOP
-                        // The 8th parameter can be anything, even an error
-                        //
-                        // ```irb
-                        // Time.utc(2022, 1, 1, 0, 0, 0, 0, StandardError)
-                        // => 2022-01-01 00:00:00 UTC
-                        // ```
+                    let arg = to_int(self, *arg)?;
+                    // unwrap is safe since to_int gaurnatees a non nil
+                    // Ruby::Integer
+                    let arg: i64 = arg.try_convert_into::<Option<i64>>(self)?.unwrap();
+
+                    match i {
+                        0 => result.year = arg,
+                        1 => result.month = arg,
+                        2 => result.day = arg,
+                        3 => result.hour = arg,
+                        4 => result.minute = arg,
+                        5 => result.second = arg,
+                        6 => result.micros = arg,
+                        7 => {
+                            // NOOP
+                            // The 8th parameter can be anything, even an error
+                            //
+                            // ```irb
+                            // Time.utc(2022, 1, 1, 0, 0, 0, 0, StandardError)
+                            // => 2022-01-01 00:00:00 UTC
+                            // ```
+                        }
+                        _ => unreachable!(),
                     }
-                    _ => unreachable!(),
                 }
+                Ok(result)
             }
-            Ok(result)
+            10 => todo!(),
+            _ => unreachable!(),
```

Comment:
is rustc not smart enough to see that this match is exhaustive. 😢 if that's the case.

Reply:
Hah indeed. It's maybe just one level of abstraction too far for it to know that args.len() and the index on the iterator are the same thing I guess :shrug: